File: TestRecursiveSum/C/TestRecursiveSum.c
//TestRecursiveSum.c
//23/03/2021
#include <stdio.h>
#include <math.h>

double recursiveSum(double array[], int arraySize){
	
	if(arraySize < 0){
		return 0;
	}
	else{
		return array[arraySize] + recursiveSum( array, arraySize-1);
	}	
};

int main(){
	
	double array[5] = {1, 3, 5, 7, 9};
	double result = 0;
	result = recursiveSum(array, sizeof(array)/sizeof(double));
	printf("The sum of array is %f", result);
	
	return 0;
}

File: TestRecursiveSum/Python/TestRecursiveSum.py
#TestRecursiveSum
#23/03/2021

#sum(array)
#Description: The function return the sum of an array
def sum(array):
	result = 0
	for i in array:
		result += i 
	return result

#recursiveSum(array)
#Description: The function return the sum of an array computed recursively
def recursiveSum(array):
	if len(array)==0:
		return 0
	else:
		return array[0] + recursiveSum(array[1:])
	
	
#TEST	
array = [1, 3, 5, 7, 9]
print(sum(array))
print(recursiveSum(array))